### ISSUE TITLE
Support dotnet build

### ISSUE THREAD

**Issue by cwensley** [picoe/Eto#1080]
Currently the Mac nugets include targets that aren't compatible with dotnet build (msbuild on .net core) as it doesn't support CodeTaskFactory.  We should pre-compile the tasks instead which should make this work.

**Comment by SteveGilham** [comment]
This is related to [MSBuild Issue #616](https://github.com/Microsoft/msbuild/issues/616); which slaims that this can be resolved by using the [RoslynCodeTaskFactory](https://www.nuget.org/packages/RoslynCodeTaskFactory)
```
<UsingTask  
    TaskName="DoNothing"  
    TaskFactory="CodeTaskFactory"  
    AssemblyFile="$(RoslynCodeTaskFactory)"
    Condition=" '$(RoslynCodeTaskFactory)' != '' ">
```

So I created a blank sample C# Eto Forms app, and added
```
    <PackageReference Include="RoslynCodeTaskFactory" Version="2.0.7" />
```
to the Desktop project, then edited `%USERPROFILE%\.nuget\packages\eto.platform.mac64\2.4.1\build\MacTemplate.targets`  to have
```
	<UsingTask TaskName="_UpdatePList" TaskFactory="CodeTaskFactory"  AssemblyFile="$(RoslynCodeTaskFactory)"
               Condition=" '$(RoslynCodeTaskFactory)' != '' ">
```
When I ran `dotnet build` in the solution directory, that yielded
```
warning : Parameters are discovered through reflection for Type="Class". Values specified in <ParameterGroup/> will be ignored. 
error MSB4065: The "IconFile" parameter is not marked for output by the "_UpdatePList" task.
```

So I removed the `<ParameterGroup> and added a `[Output]` to the `IconFile` property, and the build ran clean.  Doing `dotnet run` in the Desktop project folder then launched the app as expected.

**Comment by SteveGilham** [comment]
As an interim solution, by overriding the `_BuildAppBundle` target, this approach can be fitted into existing build processes without hacking downloaded packages.  This is working for me --

1. Add `<PackageReference Include="RoslynCodeTaskFactory" Version="2.0.7" />` to each project
2. Get amended target file `MacTemplateOverride.targets` from [this gist](https://gist.github.com/SteveGilham/1858f987d410c0c513b93ca2cc6fbace)
3. Adjust [the path to your `eto.platform.mac64` version](https://gist.github.com/SteveGilham/1858f987d410c0c513b93ca2cc6fbace#file-mactemplateoverride-targets-L34) (assumed 2.4.1 in default location) as needed
4. Add `<Import Project="<path to>\MacTemplateOverride.targets" />` at the end of each project to load the file from where you've saved it in your solution
5. `dotnet build`

**Comment by cwensley** [comment]
Now fixed with #1104.  Thanks @SteveGilham!
